pkg/manifest/manifest.go

```diff
@@ -557,36 +557,37 @@ type Dependency struct {
 	Reference string `yaml:"reference"`
 
 	// Tag is a deprecated field.  It has been replaced by Reference.
-	Tag string `yaml:"-"`
+	// This should be removed prior to v1.0.0
+	Tag string `yaml:"tag"`
 
 	Versions         []string          `yaml:"versions"`
 	AllowPrereleases bool              `yaml:"prereleases"`
 	Parameters       map[string]string `yaml:"parameters,omitempty"`
 }
 
-func (d *Dependency) Validate() error {
+func (d *Dependency) Validate(cxt *context.Context) error {
 	if d.Name == "" {
 		return errors.New("dependency name is required")
 	}
 
-	depRef := d.Reference
 	if d.Tag != "" {
-		fmt.Println("WARNING: the tag field has been deprecated in favor of reference; " +
-			"please update the Porter manifest accordingly")
-		if depRef == "" {
-			depRef = d.Tag
+		fmt.Fprintf(cxt.Out, "WARNING: the tag field for dependency %q has been deprecated "+
+			"in favor of reference; please update the Porter manifest accordingly\n",
+			d.Name)
+		if d.Reference == "" {
```

Comment:
I think the defaulting right here makes the logic in the adapter unnecessary unless there's another code path that targets?

Reply:
ditto https://github.com/getporter/porter/pull/1421#discussion_r558503530